datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -1154,12 +1154,12 @@ mod test {
         let table_scan = test_table_scan()?;
 
         let plan = LogicalPlanBuilder::from(table_scan)
-            .project(vec![lit(1) + col("a")])?
+            .project(vec![lit(1) + col("a"), col("a")])?
```

Comment:
It corresponds to `select a+1 from (select a+1 from t)`, which is not a valid query.
```sh
> select a+1 from (select a+1 from t);
Schema error: No field named a. Valid fields are "t.a + Int64(1)"
```